DisplayAllCharacters.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+session_start();
+
+$db = new PDO('mysql:host=db; dbname=lower_deck_charaters', 'root', 'password');
+
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$characterDataQuery = $db->prepare(
+    "SELECT `name`, `species`, `rank`, `image` 
+        FROM `characters`;"
+);
+
+$characterDataQuery->execute();
+
+$data = $characterDataQuery->fetchAll();
+
+foreach ($data as $characterData) {
+    echo '<div>';
+    echo '<img src="' . $characterData['image'] . '" width="120" height="200">';
+    echo '<li>' . $characterData['name'] . '</li>';
+    echo '<li>' . $characterData['species'] . '</li>';
+    echo '<li>' . $characterData['rank'] . '</li>';
+    echo '</div>';
+}
```

Comment:
You should do this under your characters h2 instead of up here.

You can just move this code down there, making sure it's in a set of <?PHP ?> tags and it should work just fine